src/main/java/bitflow4j/steps/database/WriteHistogramToElasticsearch.java

```diff
@@ -0,0 +1,89 @@
+package bitflow4j.steps.database;
+
+import bitflow4j.Sample;
+import bitflow4j.steps.BatchHandler;
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.support.replication.ReplicationResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class WriteHistogramToElasticsearch implements BatchHandler {
+
+    protected static final Logger logger = Logger.getLogger(WriteHistogramToElasticsearch.class.getName());
+
+    private final String indexName;
+    private final String identifierKey;
+    private final String identifierTemplate;
+    private final RestHighLevelClient client;
+    public WriteHistogramToElasticsearch(String host, int port1, int port2, String indexName, String identifierKey, String identifierTemplate) {
+        this.indexName = indexName;
+        this.identifierKey = identifierKey;
+        this.identifierTemplate = identifierTemplate;
+        this.client = new RestHighLevelClient(
+                RestClient.builder(
+                        new HttpHost(host, port1, "http"),
+                        new HttpHost(host, port2, "http")));
+    }
+
+    @Override
+    public List<Sample> handleBatch(List<Sample> batch) throws IOException {
+
+        IndexRequest request = new IndexRequest(indexName);
+        request.id();
+        for (int i = 0; i < batch.size(); i++) {
```

Comment:
If you don't use the loop variable i, transform this into a Java foreach loop: for (Sample sample : batch)